### DISCUSSION TITLE
Does Kitty know if a foreground CLI application wants to read user input?

### DISCUSSION THREAD

**Discussion by resolritter** [kovidgoyal/kitty#5286] [Q&A] [answered]
Hello,

While reading https://askubuntu.com/questions/1118109/how-do-i-tell-if-a-command-is-running-or-waiting-for-user-input I came up with a question: does Kitty somehow know when a CLI application wants to read user input? Even if not currently, is it _theoretically_ possible to know that?

Aside: from the foreground process' PID running in a given Kitty window, I am able to read its `/proc/$PID/syscall` (https://man7.org/linux/man-pages/man5/proc.5.html), which starts with `0 `  if the process is blocked on a `read` call, but this approach sometimes gives false positives, e.g. when the process is being piped into and not actually waiting for user input. Also it seems like `/proc/$PID/syscall` can only be read by a superuser if the process is started by a superuser, so it's not as reliable.

**Comment by kovidgoyal** [comment, marked as answer]
No, terminals can only know that the other end of the PTY device is open
in at least one process. There is no way to know if some process is
expecting to read from it. What is it you are trying to actually do?

**Reply by resolritter** [reply]
In my setup I have terminals across multiple desktop workspaces. Sometimes I don't notice when a CLI program is expecting some action out of me in a inactive workspace, for example `[sudo] password for user:`, or some other prompt.

My idea for something like this feature would be: detecting if a terminal in a inactive workspace has a foreground program which is expecting my input. With Kitty's remote control `get-text` I can read the last line of the screen and do some text-based matching that way, but it isn't something that I can apply generally.

**Comment by kovidgoyal** [comment]
I dont really see how you could pull that off. 

1) It would require continuous polling
2) There are a whole bunch of pollible syscalls a program could use,
and they are platform dependent 
3) You dont know the fd number in the program corresponding to the tty
device
4) You dont have blanket permission to access information about other
processes, even /proc could be mounted with hidepid

**Reply by resolritter** [reply]
Makes sense. IMO specially point 3 would be a pretty serious roadblock.

Thanks for entertaining the idea.
